src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -27,508 +27,303 @@ public class ResultTextMarker
-                if (componentModel == null) { return; }
-
-                ISarifLocationProviderFactory sarifLocationProviderFactory = componentModel.GetService<ISarifLocationProviderFactory>();
+            _tag.CaretEnteredTag += CaretEnteredTag;
+            _wpfTextView = wpfTextView;
+            _vsWindowFrame = vsWindowFrame;
+            _wpfTextView.Closed += this.TextViewClosed;
 
-                // Get a SimpleTagger over the buffer to color
-                _tagger = sarifLocationProviderFactory.GetTextMarkerTagger(_textView.TextBuffer);
-            }
-
-            // Add the marker
-            if (_tagger != null)
-            {
-                // The list of colors for TextMarkerTag are defined in Platform\Text\Impl\TextMarkerAdornment\TextMarkerProviderFactory.cs
-                _trackingSpan = textSnapshot.CreateTrackingSpan(span, SpanTrackingMode.EdgeExclusive);
-            }
+            return true;
         }
 
-        private bool IsValidMarker()
+        private void TextViewClosed(object sender, EventArgs e)
         {
-            return (_marker != null &&
-                    _marker.Span != null &&
-                    _marker.Span.TextBuffer != null &&
-                    _marker.Span.TextBuffer.CurrentSnapshot != null);
+            _tag.CaretEnteredTag -= CaretEnteredTag;
         }
 
-        private void SaveCurrentTrackingData(Region sourceLocation)
-        {
-            try
-            {
-                if (!IsTracking())
-                {
-                    return;
-                }
-
-                ITextSnapshot textSnapshot = _trackingSpan.TextBuffer.CurrentSnapshot;
-                SnapshotPoint startPoint = _trackingSpan.GetStartPoint(textSnapshot);
-                SnapshotPoint endPoint = _trackingSpan.GetEndPoint(textSnapshot);
+        // When the VS Editor tag has the caret moved inside of it, let's just pass along the region selection.
+        private void CaretEnteredTag(object sender, EventArgs e) => this.RaiseRegionSelected?.Invoke(this, e);
 
-                var startLine = startPoint.GetContainingLine();
-                var endLine = endPoint.GetContainingLine();
+        private static bool TryCreateTextSpanWithinDocumentFromSourceRegion(Region region, IVsWindowFrame vsWindowFrame, out TextSpan textSpan)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
 
-                var textLineStart = _textView.GetTextViewLineContainingBufferPosition(startPoint);
-                var textLineEnd = _textView.GetTextViewLineContainingBufferPosition(endPoint);
+            // SARIF regions are 1 based, VS is zero based.
+            textSpan.iStartLine = Math.Max(region.StartLine - 1, 0);
+            textSpan.iEndLine = Math.Max(region.EndLine - 1, 0);
+            textSpan.iStartIndex = Math.Max(region.StartColumn - 1, 0);
+            textSpan.iEndIndex = Math.Max(region.EndColumn - 1, 0);
```

Comment:
>; [](start = 66, length = 1)

Since the region line/column properties are guaranteed to be fully populated, none of `startLine`, `endLine`, `startColumn`, and `endColumn` can possibly be 0, so these checks aren't necessary. Just

```
textSpan.iStartLine = region.StartLine - 1;
```

_etc._ should suffice.